contracts/dev-utils/contracts/src/LibDydxBalance.sol

```diff
@@ -0,0 +1,418 @@
+        returns (uint256 solventMakerAmount)
+    {
+        solventMakerAmount = uint256(-1);
+        assert(info.actions.length >= 1);
+        IDydxBridge.BridgeAction memory withdraw = info.actions[info.actions.length - 1];
+        assert(withdraw.actionType == IDydxBridge.BridgeActionType.Withdraw);
+        int256 minCr = D18.add(_getMinimumCollateralizationRatio(info.dydx), MARGIN_RATIO_PADDING);
+        // Loop through the accounts.
+        for (uint256 accountIdx = 0; accountIdx < info.accounts.length; ++accountIdx) {
+            (uint256 supplyValue, uint256 borrowValue) =
+                _getAccountMarketValues(info, info.accounts[accountIdx]);
+            // All accounts must currently be solvent.
+            if (borrowValue != 0 && D18.div(supplyValue, borrowValue) < minCr) {
+                return 0;
+            }
+            // If this is the same account used to in the withdraw/borrow action,
+            // compute the maker amount at which it will become insolvent.
+            if (accountIdx != withdraw.accountIdx) {
+                continue;
+            }
+            // Compute the deposit/collateralization rate, which is the rate at
+            // which (USD) value is added to the account across all markets.
+            int256 dd = 0;
+            for (uint256 i = 0; i < info.actions.length - 1; ++i) {
+                IDydxBridge.BridgeAction memory deposit = info.actions[i];
+                assert(deposit.actionType == IDydxBridge.BridgeActionType.Deposit);
+                if (deposit.accountIdx == accountIdx) {
+                    dd = D18.add(
+                        dd,
+                        _getActionRateValue(
+                            info,
+                            deposit
+                        )
+                    );
+                }
+            }
+            // Compute the borrow/withdraw rate, which is the rate at which
+            // (USD) value is deducted from the account.
+            int256 db = _getActionRateValue(
+                info,
+                withdraw
+            );
+            // If the deposit to withdraw ratio is >= the minimum collateralization
+            // ratio, then we will never become insolvent at these prices.
+            if (D18.div(dd, db) >= minCr) {
+                continue;
+            }
+            // The collateralization ratio for this account, parameterized by
+            // `t` (maker amount), is given by:
+            //      `cr = (supplyValue + t * dd) / (borrowValue + t * db)`
+            // Solving for `t` gives us:
+            //      `t = (supplyValue - cr * borrowValue) / (cr * db - dd)`
+            int256 t = D18.div(
+                D18.sub(supplyValue, D18.mul(minCr, borrowValue)),
+                D18.sub(D18.mul(minCr, db), dd)
+            );
+            solventMakerAmount = LibSafeMath.min256(
+                solventMakerAmount,
+                // `t` is in maker token units, so convert it to maker wei.
```

Comment:
I may be thinking about this in totally the wrong way, but isn't `t` unitless here?

Suppose:
- Unit of `supplyValue` is `a`. The unit of `dd` is then also `a`.
- Unit of `borrowValue` is `b`. The unit of `db` is then also `b`.
- Unit of `cr` is then `a/b`. 

Find the unit of measure of `t`:
```
t = (supplyValue - cr * borrowValue) / (cr * db - dd)
→
t unit of measure = (a - (a/b) * b) / ((a/b) * b - a)
```
The units all cancel out, leaving `t` as just a scalar. 

If `t` were unitless then I think we would interpret it as the # of times we can execute the trade:
* At `t=1` we can execute the entire trade.
* At `t<1` we cannot execute the trade.
* At `t>1` we can execute the trade more than once.

So any value of `t≥1` would be valid for an order that buys `supplyValue` and sells `borrowValue`.

Reply:
Ye right. The `_toWei()` on the next line converts it back to maker units. The comment is wrong.